uploadNeuroDB/file_insertion.pl

```diff
@@ -0,0 +1,509 @@
+    exit 3; ##TODO 1: replace exit 3 by $NeuroDB::ExitCodes::MISSING_ARG
+}
+unless ( $scan_type ) {
+    print "$Usage\n\tERROR: missing -scan_type argument\n\n";
+    exit 3; ##TODO 1: replace exit 3 by $NeuroDB::ExitCodes::MISSING_ARG
+}
+unless ( $date_acquired ) {
+    print "$Usage\n\tERROR: missing -date_acquired argument\n\n";
+    exit 3; ##TODO 1: replace exit 3 by $NeuroDB::ExitCodes::MISSING_ARG
+}
+unless ( $scanner_id ) {
+    print "$Usage\n\tERROR: missing -scanner_ID argument\n\n";
+    exit 3; ##TODO 1: replace exit 3 by $NeuroDB::ExitCodes::MISSING_ARG
+}
+unless ( $coordin_space ) {
+    print "$Usage\n\tERROR: missing -coordin_space argument\n\n";
+    exit 3; ##TODO 1: replace exit 3 by $NeuroDB::ExitCodes::MISSING_ARG
+}
+unless ( $upload_id ) {
+    print "$Usage\n\tERROR: missing -upload_id argument\n\n";
+    exit 3; ##TODO 1: replace exit 3 by $NeuroDB::ExitCodes::MISSING_ARG
+}
+
+# Make sure the files specified as an argument exist and are readable
+unless (-r $file_path) {
+    print "$Usage\n\tERROR: You must specify a valid file path to insert "
+          . "using the -file_path option.\n\n";
+    ##TODO 1: replace exit 5 by $NeuroDB::ExitCodes::ARG_FILE_DOES_NOT_EXIST
+    exit 5;
+}
+# Make sure that the metadata file is readable if it is set
+if ( $metadata_file && !(-r $metadata_file) ){
+    print "\n\tERROR: The metadata file does not exist in the filesystem.\n\n";
+    ##TODO 1: replace exit 5 by $NeuroDB::ExitCodes::ARG_FILE_DOES_NOT_EXIST
+    exit 5;
+}
+
+
+
+
+###### Establish database connection
+
+my $dbh = &NeuroDB::DBI::connect_to_db(@Settings::db);
+
+
+
+
+###### Get config settings
+
+my $data_dir = NeuroDB::DBI::getConfigSetting(\$dbh, 'dataDirBasepath');
+
+
+
+
+###### For the log, temp directories and notification spools
+
+# determine local time
+my ($sec,$min,$hour,$mday,$mon,$year,$wday,$yday,$isdst) = localtime(time);
+my $today = sprintf( "%4d-%02d-%02d %02d:%02d:%02d",
+                     $year+1900,$mon+1,$mday,$hour,$min,$sec
+);
+
+# create the temp directory
+my $template = "FileLoad-$hour-$min-XXXXXX";
+my $TmpDir   = tempdir($template, TMPDIR => 1, CLEANUP => 1 );
+my @temp     = split(/\//, $TmpDir);
+my $temp_log = $temp[$#temp];
+
+# create the log file
+my $log_dir  = $data_dir . "/logs";
+my $log_file = $log_dir . "/" . $temp_log . ".log";
+my $message  = "\nlog dir is $log_dir and log file is $log_file.\n";
+print $message if $verbose;
+
+# open log file and write successful connection to DB
+open( LOG, ">>", $log_file ) or die "\nError Opening $log_file.\n";
+LOG->autoflush(1);
+&logHeader();
+$message = "\n==> Successfully connected to database\n";
+print LOG $message;
+
+# create Notify and Utility objects
+my $notifier = NeuroDB::Notify->new(\$dbh);
+my $utility = NeuroDB::MRIProcessingUtility->new(
+    \$dbh, 0, $TmpDir, $log_file, $verbose
+);
+
+
+
+
+##### Verify that the provided scanner ID refers to a valid scanner entry
+(my $query = <<QUERY) =~ s/\n/ /gm;
```

Comment:
Why not just do 

`my $query = 'SELECT ID FROM mri_scanner WHERE ID=?';`

I think the `<<` operator should be reserved for true multi-line strings.

Reply:
You're right. Changing all of them.